semeio/communication/reporter.py

```diff
@@ -0,0 +1,37 @@
+import json
+import os
+
+
+class FileReporter(object):
+    def __init__(self, output_dir):
+        self._output_dir = output_dir
+
+        if not os.path.isabs(output_dir):
+            err_fmt = "Expected output_dir to be an absolute path, received '{}'"
+            raise ValueError(err_fmt.format(output_dir))
+
+    def publish(self, namespace, data):
+        output_file = self._prepare_output_file(namespace) + ".json"
+
+        with open(output_file, "w") as f:
+            json.dump(data, f)
+
+    def publish_msg(self, namespace, msg):
+        fmt = "{}\n"
+        output_file = self._prepare_output_file(namespace)
+        with open(output_file, "a") as f:
+            f.write(fmt.format(msg))
```

Comment:
should message be stripped for trailing new line, in order to prevent empty lines?

Reply:
I'm into just giving people what they're asking for. This is unstructured output anyway. If you want trailing spaces, you can have them? 🤷